providers/providers.go

```diff
@@ -47,6 +52,7 @@ func New(provider string, p *ProviderData) Provider {
 	case "digitalocean":
 		return NewDigitalOceanProvider(p)
 	default:
+		logger.Errorf("unknown provider '%s', defaulting to 'google'", provider)
```

Comment:
I like where you're thinking:

But I think if we truly get an unknown provider, we should fail at the validation stage and not startup rather than defaulting to google and making an error message.

This is a little out of scope from this PR though, we should move that fix out of this one.

Reply:
This was the original behavior, I didn't want to be too intrusive in case there was a reason for it.

Reply:
Can you jot down this finding in another issue and we'll handle it in another PR? There's some aspects of the proper fix for this that are worth discussing that are out of scope from this PR.

Reply:
Sure.